File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        
        longest = 0

        l = 0
        n = len(blocks)
        window_size = k
        blacks = 0
        for i in range(k):
            if blocks[i] == 'B':
                blacks += 1
        
        best = k - blacks
        
        l = 0
        for r in range(k, n):
            if blocks[r] == 'B':
                blacks += 1
            
            if blocks[l] == 'B':
                blacks -= 1
            l += 1
            
            best = min(best, k - blacks)
        return best
        
            

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int n = blocks.length();

        int blacks = 0;
        for (int i = 0; i < k; i++) {
            if (blocks.charAt(i) == 'B') {
                blacks++;
            }
        }
        int best = k - blacks;
        int l = 0;
        for (int r = k; r < n; r++) {
            if (blocks.charAt(r) == 'B') {
                blacks++;
            }
            if (blocks.charAt(l) == 'B') {
                blacks--;
            }
            l++;
            
            best = Math.min(best, k - blacks);
        }

        return best;
    }
}